ballerina/tests/stream_test.bal

```diff
@@ -189,9 +222,27 @@ public function testStreamJson() returns Error? {
 
 @test:Config {}
 public function testStreamJsonOnOpen() returns Error? {
-    Client wsClient = check new("ws://localhost:21402/onJsonStreamOnOpen/");
+    Client wsClient = check new ("ws://localhost:21402/onJsonStreamOnOpen/");
     json data = check wsClient->readMessage();
     test:assertEquals(data, {"x": 1, "y": 2});
     json data2 = check wsClient->readMessage();
     test:assertEquals(data2, {"x": 4, "y": 5});
 }
+
+@test:Config {}
+public function testConcurrentRequestDuringStreamResponse() returns Error? {
+    Client wsClient = check new ("ws://localhost:21402/onConcurrentRequest/");
+    check wsClient->writeMessage({event: "subscribe"});
+    int data = check wsClient->readMessage();
+    test:assertEquals(data, 1);
+    check wsClient->writeMessage("Hello");
+    while true {
+        int res = check wsClient->readMessage();
+        if res >= 10 {
+            test:assertFail("Did not receive -1 as the response from the service");
```

Comment:
If we do the assertFail here, there is a possibility that this test may fail if the service responds with the next message of the stream before it returns the `-1` from the `onMessage`. Better to remove this `if res >= 10 {` and include `test:assertFail` after the `while true`. WDYT?